### PR TITLE
@lexical/playground: Moved Vite config to TS and removed code duplication

### PR DESCRIPTION
**Pull request by StyleT**
Here we extract module resolution config to `viteModuleResolution.ts` and move both Vite configs from `.js` to `.ts`

### CODE REVIEW

**Review by ivailop7 — APPROVED** [review]
LGTM, I do think we can have the vite config as .mts (https://vitejs.dev/guide/troubleshooting#config) if we want to move more towards ESM practices.

**Comment by ivailop7** [comment]
Bumping the vite to v5 in package.json has been in my do list for a while. In case you want to give it a go.

**Review by ivailop7 — APPROVED** [review]
@StyleT I think this is good to be merged

File: packages/lexical-playground/vite.prod.config.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  },
+  define: {
+    'process.env.IS_PREACT': process.env.IS_PREACT,
+  },
+  plugins: [
+    replaceCodePlugin({
+      replacements: [
+        {
+          from: /__DEV__/g,
+          to: 'true',
```
Comment (canastro) [inline packages/lexical-playground/vite.prod.config.ts:42]: should this be false?

**Comment by etrepum** [comment]
This reverted the /esm/ example in the playground from #5618, I'll create another PR to restore it

**Pull request merged**